### PR TITLE
Testing citation data/date selection

### LINKED ISSUE #573: [Testing] `packages/website/src/features/citation/ui/citation-data-category-selection.tsx`
### User Story

As a developer, I want the codebase to be fully tested, so that I am sure that we are promoting high code quality by releasing stable code with no regressions.

### Description

This work, along with many other tickets, is to improve our testing code coverage by writing tests for the file in the title.

To view our current code coverage, run `pnpm test`. Then, open this file: `<project-root>/coverage/lcov-report/index.html`.

### Acceptance Criteria

1. The file in the title is fully tested, including statements, branches, functions, and lines.
2. All existing test suites pass.

### Technical References

- [Jest documentation](https://jestjs.io/docs/getting-started)
- [React Testing Library documentation](https://testing-library.com/docs/react-testing-library/intro/)
- Look into existing test suites

### Additional Information

N/A

### PR DESCRIPTION
**Pull request by bennyv8**
### Description

Adds tests files for Citation-Data-Category-Selection and Citation-Date-Presets-Selection
- Tests for component rendering
- Tests for user selection of options using provided constants

### Related Issues

Resolves #573 #576 

### Testing

- Ran testing suites.

### Checklist

- [X] I have followed all conventions outlined in our documentation.
- [X] I have fully tested my changes and confirmed that all new and existing tests pass.
- [X] I have written meaningful commit messages for all changes.
- [X] I have linted and formatted my changes to follow the code style of this repository.
- [ ] I have updated our documentation, accordingly.
- [X] I have checked currently opened pull requests to ensure that there are no pending pull request for the same changes or issues.
- [X] I have confirmed that this pull request fully meets the acceptance criteria for all related issues listed above.

### CODE REVIEW

File: packages/website/src/features/citation/ui/citation-data-category-selection.test.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  test("renders citation data categories with initial placeholder text", async () => {
+    expect(screen.getByText(options[0]));
+  });
+
+  test("should allow user to change the current citation data category", async () => {
+    const user = userEvent.setup();
+
+    expect(screen.queryByText("Total $ Fine Amount")).toBe(null);
+    await user.click(screen.getByRole("combobox"));
+    await user.click(screen.getByText(options[1]));
```
Comment (gibsonliketheguitar) [inline packages/website/src/features/citation/ui/citation-data-category-selection.test.tsx:23]: see comment above

File: packages/website/src/features/citation/ui/citation-date-presets-selection.test.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { RelativeDatePresets } from "@/shared/lib/constants/date";
+import "@testing-library/jest-dom";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+
+import CitationDatePresetsSelection from "./citation-date-presets-selection";
+
+describe("Citation Date Presets Selection", () => {
+  const options = Object.values(RelativeDatePresets);
```
Comment (gibsonliketheguitar) [inline packages/website/src/features/citation/ui/citation-date-presets-selection.test.tsx:9]: see comment above

File: packages/website/src/features/citation/ui/citation-date-presets-selection.test.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  });
+
+  test("renders citation date presets with initial placeholder text", async () => {
+    expect(screen.getByText(options[0]));
+  });
+
+  test("should allow user to change the current citation date preset", async () => {
+    const user = userEvent.setup();
+
+    expect(screen.queryByText("Total $ Fine Amount")).toBe(null);
```
Comment (gibsonliketheguitar) [inline packages/website/src/features/citation/ui/citation-date-presets-selection.test.tsx:21]: This maybe a redundant test, and incorrect. We should be testing "Past Year". In addition this type should be in pick list, and it test should be specific and broken up into pieces.

A better implementation maybe
```
   describe,  Citation-date-present-selection inter
      test "should not render default option"
      test " should render selected option"
```

what we really want to is to test the copy
```
{
  "YEARS_1" = "Past 1 Year",
  "YTD" = "Year to Date",
  "MONTHS_3" = "3 Months",
  "MONTHS_1" = "1 Month",
  // "YEARS_ALL" = "All Years",
}
```

**Review by gibsonliketheguitar — CHANGES_REQUESTED** [review]
Great work. Thank you for writing test! 

Just a few more test to go! 

One thing i do want to mention is, in the future: [Best practice of PRs](https://www.atlassian.com/blog/git/written-unwritten-guide-pull-requests). Please break up PRs into their own context. 2 tests 2 Pr.

In addition Can we also get copy of what test were ran, and a table of the new test coverage. 
[Example
](https://github.com/hackforla/lucky-parking/pull/541)

File: packages/website/src/features/citation/ui/citation-date-presets-selection.test.tsx (hunk 3 above)
Comment (bennyv8) [reply packages/website/src/features/citation/ui/citation-date-presets-selection.test.tsx:21]: Hi Gibson, thanks for the feedback. My bad for combining the two PR's, will be sure to follow best practices next time. 

I refactored the tests to address the issues brought up and tried to re-shift the focus on the implementation of the actual citation data categories and citation date presets.

- I tested for rendering of all the citationDataCategories and citationDatePresets options.
- I broke up the tests into two describe blocks: one for rendering and the other for interactions.
- I made all the text tests explicit to the data categories and date presets.
- I also added a test to ensure that the components return the correct option.
 
 
Let me know what you think
 
### Testing

Copy of tests
```
 PASS  src/features/citation/ui/citation-date-presets-selection.test.tsx
  Citation-date-presets-selection rendering
    ✓ should render citation date presets with initial placeholder (104 ms)
    ✓ should render all citation date presets options (401 ms)
  Citation-date-presets-selection interactions
    ✓ should not render with default option (184 ms)
    ✓ should render with selected option (187 ms)
    ✓ should return selected option (184 ms)
    
 PASS  src/features/citation/ui/citation-data-category-selection.test.tsx
  Citation-data-category-selection rendering
    ✓ should render citation data categories with initial placeholder (109 ms)
    ✓ should render all citation data categories options (474 ms)
  Citation-data-category-selection interactions
    ✓ should not render with default option (199 ms)
    ✓ should render with selected option (213 ms)
    ✓ should return selected option (192 ms)
    
    Test Suites: 2 passed, 2 total
Tests:       10 passed, 10 total
Snapshots:   0 total
Time:        7.229 s
Ran all test suites matching /citation/i.
```

Copy of coverage

File                                           | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
-----------------------------------------------|---------|----------|---------|---------|-------------------                             |                  
 src/features/citation/lib                     |     100 |      100 |     100 |     100 |                   
  constants.ts                                 |     100 |      100 |     100 |     100 |                   
  index.ts                                     |     100 |      100 |     100 |     100 |    
  .               
 src/features/citation/ui                      |   57.14 |        0 |   57.14 |   53.84 |                   
  citation-data-category-selection.tsx         |     100 |      100 |     100 |     100 |              
  citation-date-presets-selection.tsx          |     100 |      100 |     100 |     100 |              
  .     
 src/shared/lib/constants                      |     100 |      100 |     100 |     100 |                   
  date.ts                                      |     100 |      100 |     100 |     100 |                        
-----------------------------------------------|---------|----------|---------|---------|-------------------

**Review by gibsonliketheguitar — APPROVED** [review]
Looks great! Awesome work.

**Pull request merged**